src/contracts/strategies/StrategyFactory.sol

```diff
@@ -0,0 +1,105 @@
+ *        and automatically adding them to the StrategyWhitelist in EigenLayer.
+ * @author Layr Labs, Inc.
+ * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
+ * @dev This may not be compatible with non-standard ERC20 tokens. Caution is warranted.
+ */
+contract StrategyFactory is StrategyFactoryStorage, OwnableUpgradeable, Pausable {
+    uint8 internal constant PAUSED_NEW_STRATEGIES = 0;
+
+    /// @notice EigenLayer's StrategyManager contract
+    IStrategyManager public immutable strategyManager;
+
+    /// @notice Since this contract is designed to be initializable, the constructor simply sets the immutable variables.
+    constructor(IStrategyManager _strategyManager) {
+        strategyManager = _strategyManager;
+        _disableInitializers();
+    }
+
+    function initialize(
+        address _initialOwner,
+        IPauserRegistry _pauserRegistry,
+        uint256 _initialPausedStatus,
+        IBeacon _strategyBeacon
+    ) public virtual initializer {
+        _transferOwnership(_initialOwner);
+        _initializePauser(_pauserRegistry, _initialPausedStatus);
+        _setStrategyBeacon(_strategyBeacon);
+    }
+
+    /**
+     * @notice Deploy a new strategyBeacon contract for the ERC20 token.
+     * @dev A strategy contract must not yet exist for the token.
+     * $dev Immense caution is warranted for non-standard ERC20 tokens, particularly "reentrant" tokens
+     * like those that conform to ERC777.
+     */
+    function deployNewStrategy(IERC20 token)
+        external
+        onlyWhenNotPaused(PAUSED_NEW_STRATEGIES)
+        returns (IStrategy newStrategy)
+    {
+        require(!isBlacklisted[token], "StrategyFactory.deployNewStrategy: Token is blacklisted");
+        require(
+            tokenStrategy[token] == IStrategy(address(0)),
+            "StrategyFactory.deployNewStrategy: Strategy already exists for token"
+        );
+        IStrategy strategy = IStrategy(
+            address(
+                new BeaconProxy(
+                    address(strategyBeacon),
+                    abi.encodeWithSelector(StrategyBase.initialize.selector, token, pauserRegistry)
+                )
+            )
+        );
+        _setStrategyForToken(token, strategy);
+        IStrategy[] memory strategiesToWhitelist = new IStrategy[](1);
+        bool[] memory thirdPartyTransfersForbiddenValues = new bool[](1);
+        strategiesToWhitelist[0] = strategy;
+        thirdPartyTransfersForbiddenValues[0] = false;
+        strategyManager.addStrategiesToDepositWhitelist(strategiesToWhitelist, thirdPartyTransfersForbiddenValues);
+        return strategy;
+    }
+
+    /**
+     * @notice Owner-only function to prevent strategies from being created for given tokens.
+     * @param tokens An array of token addresses to blacklist.
+     */
+    function blacklistTokens(IERC20[] calldata tokens) external onlyOwner {
```

Comment:
Is the plan to blacklist already deployed tokens via the initializer? We don't want to get frontrun on a permissionless deployment

Reply:
Resolved offline -> the deployment requires us to wait to transfer the `strategyWhitelister` role so we won't need to add this functionality in the initializer